.github/workflows/ci.yml

```diff
@@ -191,88 +187,90 @@ jobs:
           ./zcutil/build.sh
           -j"${{ steps.nproc.outputs.count }}"
 
-      - name: Authenticate to Google Cloud
-        uses: google-github-actions/auth@v2
-        with:
-          credentials_json: ${{ secrets.GCP_SA_KEY }}
-
       - name: Upload zcashd artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/zcashd{0}', matrix.file_ext) }}-${{ matrix.name }}
```

Comment:
```suggestion
          name: ${{ format('zcashd{0}', matrix.file_ext) }}-${{ matrix.name }}
```

Reply:
According to the [docs](https://github.com/actions/upload-artifact?tab=readme-ov-file#usage) this must be a file name, not a path.

If the source only contains files in a subdirectory, the path prefix [will be stripped, unless it contains a wildcard](https://github.com/actions/upload-artifact?tab=readme-ov-file#upload-using-multiple-paths-and-exclusions). However, in this case I don't think it matters. If my mental model is correct, artifacts are in a flat namespace, not a hierarchical one. Uploading more than one file creates/updates a ZIP file named according to the artifact name. So given that we are only uploading individual files to differently-named artifacts, losing the path doesn't matter. (If it did, there is an ugly workaround [here](https://github.com/actions/upload-artifact/issues/344#issuecomment-1379232156), or I think you can also work around it by using a wildcard like `*src`, say.)